File: app/src/main/java/com/letianpai/robot/desktop/apps/adapter/AppListItem.java
package com.letianpai.robot.desktop.apps.adapter;

import android.graphics.Bitmap;

import java.util.Objects;

/**
 * @author liujunbin
 */
public class AppListItem {
    public static final int APP_START_PACKAGE = 0;
    public static final int APP_START_ACTION = 1;

    private int appIcon;            //图标；
    private int appNameTextColor;   //生成文字颜色；
    private String appName;         //应用名字；
    private String appPackageName;  //包名；
    private String appTag;          //程序Tag；
    private String appDescription;  //程序描述
    private Bitmap appBitmap;       //
    private int appIconSmall;            //图标；
    private Bitmap appBitmapSmall;       //
    private int appBg;
    private int startType;          //应用启动的方式
    private String startValue;         //应用启动的相关值

    public int getAppIcon() {
        return appIcon;
    }

    public void setAppIcon(int appIcon) {
        this.appIcon = appIcon;
    }

    public String getAppName() {
        return appName;
    }

    public void setAppName(String appName) {
        this.appName = appName;
    }

    public void setAppPackageName(String appPackageName) {
        this.appPackageName = appPackageName;
    }

    public String getAppPackageName() {
        return appPackageName;
    }

    public String getAppTag() {
        return appTag;
    }

    public void setAppTag(String appTag) {
        this.appTag = appTag;
    }

    public void setAppNameTextColor(int appNameTextColor) {
        this.appNameTextColor = appNameTextColor;
    }

    public int getAppNameTextColor() {
        return appNameTextColor;
    }


    public Bitmap getAppBitmap() {
        return appBitmap;
    }

    public void setAppBitmap(Bitmap appBitmap) {
        this.appBitmap = appBitmap;
    }

    public void setAppIconSmall(int appIconSmall) {
        this.appIconSmall = appIconSmall;
    }

    public int getAppIconSmall() {
        return appIconSmall;
    }

    public Bitmap getAppBitmapSmall() {
        return appBitmapSmall;
    }

    public void setAppBitmapSmall(Bitmap appBitmapSmall) {
        this.appBitmapSmall = appBitmapSmall;
    }

    public int getAppBg() {
        return appBg;
    }

    public void setAppBg(int appBg) {
        this.appBg = appBg;
    }

    public int getStartType() {
        return startType;
    }

    public void setStartType(int startType) {
        this.startType = startType;
    }

    public String getStartValue() {
        return startValue;
    }

    public void setStartValue(String startValue) {
        this.startValue = startValue;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        AppListItem that = (AppListItem) o;
        return appPackageName.equals(that.appPackageName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(appPackageName);
    }

    @Override
    public String toString() {
        return "AppListItem{" +
                "appIcon=" + appIcon +
                ", appNameTextColor=" + appNameTextColor +
                ", appName='" + appName + '\'' +
                ", appPackageName='" + appPackageName + '\'' +
                ", appTag='" + appTag + '\'' +
                ", appDescription='" + appDescription + '\'' +
                ", appBitmap=" + appBitmap +
                ", appIconSmall=" + appIconSmall +
                ", appBitmapSmall=" + appBitmapSmall +
                ", appBg=" + appBg +
                ", startType=" + startType +
                ", startValue='" + startValue + '\'' +
                '}';
    }
}

File: app/src/main/java/com/letianpai/robot/desktop/apps/adapter/AppListItem.kt
package com.letianpai.robot.desktop.apps.adapter

import android.graphics.Bitmap
import java.util.Objects

/**
 * @author liujunbin
 */
class AppListItem {
    var appIcon: Int = 0
    var appNameTextColor: Int = 0
    var appName: String? = null
    var appPackageName: String? = null
    var appTag: String? = null
    private val appDescription: String? = null
    var appBitmap: Bitmap? = null
    var appIconSmall: Int = 0
    var appBitmapSmall: Bitmap? = null
    var appBg: Int = 0
    var startType: Int = 0 //Application start-up method
    var startValue: String? = null //Application start-up related values


    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as AppListItem
        return appPackageName == that.appPackageName
    }

    override fun hashCode(): Int {
        return Objects.hash(appPackageName)
    }

    override fun toString(): String {
        return "AppListItem{" +
                "appIcon=" + appIcon +
                ", appNameTextColor=" + appNameTextColor +
                ", appName='" + appName + '\'' +
                ", appPackageName='" + appPackageName + '\'' +
                ", appTag='" + appTag + '\'' +
                ", appDescription='" + appDescription + '\'' +
                ", appBitmap=" + appBitmap +
                ", appIconSmall=" + appIconSmall +
                ", appBitmapSmall=" + appBitmapSmall +
                ", appBg=" + appBg +
                ", startType=" + startType +
                ", startValue='" + startValue + '\'' +
                '}'
    }

    companion object {
        const val APP_START_PACKAGE: Int = 0
        const val APP_START_ACTION: Int = 1
    }
}
